src/main/java/org/folio/edge/orders/ValidateHandler.java

```diff
@@ -26,7 +26,22 @@ protected void handle(RoutingContext ctx) {
         (client, params) -> {
           PurchasingSystems ps = PurchasingSystems.fromValue(params.get(PARAM_TYPE));
           logger.info("Request is from purchasing system: " + ps.toString());
-          ((OrdersOkapiClient) client).validate(
+          ((OrdersOkapiClient) client).validate(HttpMethod.GET,
+              ps,
+              ctx.request().headers(),
+              resp -> handleProxyResponse(ps, ctx, resp),
+              t -> handleProxyException(ctx, t));
+        });
+  }
+
+  protected void handlePost(RoutingContext ctx) {
```

Comment:
Instead of defining a whole new method specific for this purpose you could get the HTTP method from the routing context and pass that into the validate method...

Reply:
yup I wanted to do that and overlooked the availability of request.method(). Thanks